### PR TITLE
Adds a WordPress.org theme install menu item.

### PR DESCRIPTION
**Pull request by cpapazoglou**
Fixes https://github.com/Automattic/wp-calypso/issues/49238

#### Changes proposed in this Pull Request:

* Adds a WordPress.org theme install menu item for WoA sites only so users have the chance to install a theme from the WordPress.org repository

#### Testing instructions:

* Select this branch in Jetpack Beta
* Make sure that a Appearance -> WordPress.org Themes exists under Themes 
* The above menu redirects to `/wp-admin/theme-install.php`
* Run `yarn docker:phpunit --filter Test_Atomic_Admin_Menu` and confirm there are no failures

Before | After
-------|------
 | 

#### Proposed changelog entry for your changes:

* Adds a WordPress.org theme install menu item for Atomic sites.

### CODE REVIEW

**Comment by cpapazoglou** [comment]
@obenland, @sfougnier any preference for the wording ("WordPress.org Themes") or the order of this menu item?

**Comment by obenland** [comment]
How about "Add New Theme"?

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php
Comment (jeherve) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php:216]: Any specific reason we can't go with just "Themes"? I wonder if "WordPress.org" here could be more confusing than anything else.

Comment (cpapazoglou) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php:216]: We already have a "Themes" menu `https://wordpress.com/themes/[DOMAIN]`.

**Comment by cpapazoglou** [comment]
> How about "Add New Theme"?

I opted for "WordPress.org Themes" because:

1) Adding a premium WordPress.com theme cannot be accomplished there. You are still required to go through `https://wordpress.com/themes/[DOMAIN]` (Correct me if I am wrong here)
2) Except from adding a theme, it also allows users to activate a WordPress.org theme

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php
Comment (cpapazoglou) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php:216]: I see your point about the "confusion" though!

**Review by getdave — CHANGES_REQUESTED** [review]
I tested this and unfortunately I think `WordPress.org` is confusing. Our users don't understand the difference at the best of times.

Please can we go with Konstantin's suggestion of `Add New Theme`? If it needs to be amended later we can do that.

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php
Comment (getdave) [reply projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php:216]: See https://github.com/Automattic/jetpack/pull/18780#pullrequestreview-588420913

**Comment by mmtr** [comment]
Just pushed out an update renaming the menu item to `Add New Theme` ([so it's aligned with the naming used in Core](https://core.trac.wordpress.org/browser/trunk/src/wp-admin/themes.php?rev=49969#L230)). Also changed the checked capability to `install_themes`, since [that's the capability used in Core too](https://core.trac.wordpress.org/browser/trunk/src/wp-admin/themes.php?rev=49969#L224).

**Review by getdave — APPROVED** [review]
Tested this and it worked as described. Nice work @cpapazoglou and @mmtr.

Screen Shot 2021-02-11 at 14 52 05

**Review by mmtr — APPROVED** [review]

**Comment by mmtr** [comment]
Looks like I cannot unblock merging this PR until https://github.com/Automattic/jetpack/pull/18792 is merged, so I'm adding back the needs review column for getting an approval from Crew

**Review by jeherve — APPROVED** [review]

**Comment by jeherve** [comment]
Cherry-picked to `jetpack/branch-9.4` in c04c7e8388d739787c46dcdc3558b00df35aabc7

**Comment by mmtr** [comment]
Deployed to WP.com in r220922-wpcom

**Pull request merged**
